server/src/main/java/org/elasticsearch/health/metadata/HealthMetadataService.java

```diff
@@ -124,39 +130,34 @@ private void enable(boolean enabled) {
             resetHealthMetadata("health-node-enabled");
         } else {
             clusterService.removeListener(clusterStateListener);
-            readyToPublish = false;
+            readyToPublishAsMaster = false;
         }
     }
 
     private void updateOnClusterStateChange(ClusterChangedEvent event) {
         final boolean wasMaster = event.previousState().nodes().isLocalNodeElectedMaster();
         isMaster = event.localNodeMaster();
         if (isMaster && wasMaster == false) {
```

Comment:
Wait, this can't be right.... When we have `isMaster == false` then this `if` statement will be `true` and it will try to publish, right?

Reply:
if `isMaster == false` then it will short-circuit and fall to the following `else-if`, where the condition will be true, making `readyToPublish = false` . I don't think I fully followed your question

Reply:
Disclaimer, I did write it, but now reading it.... it doesn't make sense :p

Reply:
ignore..... I added parentheses in my mind